traits/has_traits.py

```diff
@@ -341,6 +343,37 @@ def handler_getter(instance, name):
     )
 
 
+def _parse_expression(expression):
+    """ Parse a given expression or list of expressions to convert them into
+    ObserveExpression objects if needed.
+
+    Parameters
+    ----------
+    expression : str or list or ObserverExpression
+    A description of what traits are being observed.
+    If this is a list, each item must be a string or an Expression.
+
+    Returns
+    -------
+    expressions : list
```

Comment:
For lists, dictionaries and the like, it's useful to describe the type of the contents in the docstring

```suggestion
    expressions : list of ObserverExpression
```